app.js

```diff
@@ -249,44 +277,292 @@ const app = Vue.createApp({
             return eq1 + eq2 + eq3 + eq4 + eq5 +eq6
         },
         baseScore() {
+            //define lookup table
+            lookuptable = this.currentLookup;
+
+            AV_levels={"N": 0.0, "A": 0.1, "L": 0.2, "P": 0.3}
+            PR_levels={"N": 0.0, "L": 0.1, "H": 0.2}
+            UI_levels={"N": 0.0, "P": 0.1, "A": 0.2}
+
+            AC_levels={'L':0.0, 'H':0.1}
+            AT_levels={'N':0.0, 'P':0.1}
+        
+            VC_levels={'H':0.0, 'L':0.1, 'N':0.2}
+            VI_levels={'H':0.0, 'L':0.1, 'N':0.2}
+            VA_levels={'H':0.0, 'L':0.1, 'N':0.2}    
+
+            SC_levels={'H':0.1, 'L':0.2, 'N':0.3}
+            SI_levels={'S':0.0, 'H':0.1, 'L':0.2, 'N':0.3}
+            SA_levels={'S':0.0, 'H':0.1, 'L':0.2, 'N':0.3}
+
+            CR_levels={'H':0.0, 'M':0.1, 'L':0.2}
+            IR_levels={'H':0.0, 'M':0.1, 'L':0.2}
+            AR_levels={'H':0.0, 'M':0.1, 'L':0.2}
+
+            E_levels={'U': 0.2, 'P': 0.1, 'A': 0}
+
+
             lookup = this.macroVector
             // Exception for no impact on system
             if(lookup.includes("33")) {
                 return "0.0"
             }
-            value = this.cvssLookupData[lookup]
+            value = lookuptable[lookup]
+
+            eq1_val = parseInt(lookup[0])
+            eq2_val = parseInt(lookup[1])
+            eq3_val = parseInt(lookup[2])
+            eq4_val = parseInt(lookup[3])
+            eq5_val = parseInt(lookup[4])
+            eq6_val = parseInt(lookup[5])
+
+            //compute next lower macro, it can also not exist
+            eq1_next_lower_macro = "".concat(eq1_val+1,eq2_val,eq3_val,eq4_val,eq5_val,eq6_val)
```

Comment:
Variable names suggest we should use an array or some better data structure? Hard to judge without proper computational requirements.